server/services/professionalContactsService.ts

```diff
@@ -156,27 +176,38 @@ function mapCommunityManagerToProfessionalContact(communityManager: CommunityMan
     relationshipDescription: 'Community Offender Manager',
     relationship: 'COM',
     unallocated,
-  }
+  })
 }
 
-function mapKeyWorkerToProfessionalContact(keyWorker: KeyWorker): ProfessionalContact {
-  const { firstName, lastName, email } = keyWorker
-
-  return {
-    firstName,
-    lastName,
-    emails: [email].filter(Boolean),
-    phones: [],
-    address: undefined,
-    relationshipDescription: 'Key Worker',
-    relationship: 'KW',
-  }
+function mapKeyWorkerToProfessionalContact(
+  keyWorkerResult: Result<KeyWorker>,
+): Result<ProfessionalContact, ProfessionalContactApiError>[] {
+  if (keyWorkerResult.isFulfilled() && keyWorkerResult.getOrThrow() === null) return []
+
+  const keyworkerRelationship = { relationship: 'KW', relationshipDescription: 'Key Worker' }
+
+  return [
+    keyWorkerResult.map(
+      keyWorker => ({
+        ...keyworkerRelationship,
+        firstName: keyWorker.firstName,
+        lastName: keyWorker.lastName,
+        emails: [keyWorker.email].filter(Boolean),
+        phones: [],
+        address: undefined,
+      }),
+      _error => keyworkerRelationship,
```

Comment:
Can now map the error to decorate with extra details such as which contact failed